climate/plugins/oshosts/host_plugin.py

```diff
@@ -183,18 +190,25 @@ def update_reservation(self, reservation_id, values, usage_enforcement=False, us
         hosts_in_pool = pool.get_computehosts(
             reservation['resource_id'])
 
+        host_allocations = db_api.host_allocation_get_all_by_values(reservation_id=reservation_id)
+
+
         # Check if we have enough available SUs for update
         if usage_enforcement:
+            total_su_factor = sum(
```

Comment:
If the hosts change (I thought we couldn't do this?) then there'd need to be a backout step where encumbered SUs are released for whatever the remainder of the reservation was (`release(old_total_su_factor * (end_date - now))`), then recomputed for what it is going forward (`encumber(new_total_su_factor * (end_date - now))`).